src/wallet/spend.h

```diff
@@ -121,6 +121,26 @@ std::optional<SelectionResult> AttemptSelection(const CWallet& wallet, const CAm
 std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, const CAmount& nTargetValue, const CoinEligibilityFilter& eligibility_filter, const std::vector<COutput>& available_coins,
                         const CoinSelectionParams& coin_selection_params);
 
+// User manually selected inputs that must be part of the transaction
+struct PreSelectedInputs
```

Comment:
We actually had this convo at the beginning of the PR (https://github.com/bitcoin/bitcoin/pull/25685#discussion_r938067217).

I implemented it two/three months ago and the end result wasn't so satisfactory.

If my memory doesn’t fail me, the reasons were:

1. We would be mixing responsibilities if we do that. The `SelectionResult` class is particular to the coin selection process (in the future, it might even be decoupled into a separated library) and `FetchSelectedInputs` is a wallet method that we use to obtain a set of outputs (similar to `AvailableCoins` but instead of traversing the whole wallet tx map, it fetches the specific outputs directly), which could be used in different contexts and not only inside the tx creation process (places where the `SelectionResult` class wouldn't make much sense).

2. The `SelectionResult` constructor requires a target, which we don’t have up until we fetch and sum-up all the manually selected outputs. So, we would end up doing the same that we do inside `SelectCoins` but inside `FetchSelectedInputs`: (1) first fetch and store all the outputs in a vector, then at the end of the method, create a `SelectionResult` with the accurate target and return it (so there would really be no difference with the current approach unless we add another constructor and a new `AddInput` method inside `SelectionResult` that keeps track of target etc or.. pass the tx target to `FetchSelectedInputs` which I think that is ugly as the tx target has nothing to do with a fetching-only function). 
3. Plus, the new struct caches the total amount and contemplates the effective/raw amount distinction. Which, for example, made #26203 issue a non-issue here.
.

With this I’m not saying that we should marry to this struct (we could also use `CoinsResult` instead) but I don’t think that changing it to `SelectionResult` will make the code better (at least for now). It would probably just continue entangling the coin selection code inside wallet general functions that we should be able to re-use in other processes.

Plus, after the next PR in the line #25806, I think that this process will be in a much better shape to re-do this convo once more. Mainly because the coin selection process will not be so attached to the wallet internals. So, we might be able to decouple it into a different module or an external library.